caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -330,9 +330,15 @@ def train(
         else:
             error.value_check(
                 "<FPT36947542E>",
-                tuning_config.output_model_types in base_model.PROMPT_OUTPUT_TYPES,
+                len(
+                    list(
+                        set(tuning_config.output_model_types)
+                        & set(base_model.PROMPT_OUTPUT_TYPES)
+                    )
+                )
+                == len(tuning_config.output_model_types),
```

Comment:
Does this solution have issue if someone passes `output_model_types` as `List[str]` instead of `enum` ? since the `TuningConfig` datamodel accepts `List[str]` as data-type for `output_model_types`, I think we would need to explicitly convert it to enum for this validation and for further processing in this function.

Reply:
I was thinking something like this:
```python
            # Change tuning config to PromptOutputModelType
            output_model_types = []
            for output_type in tuning_config.output_model_types:
                 output_model_types.append(PromptOutputModelType(output_type))

            error.value_check(
                "<FPT36947542E>",
                all(
                    [
                        output_type in base_model.PROMPT_OUTPUT_TYPES
                        for output_type in  output_model_types
                    ]
                ),
                "{} not supported for base model type {}".format(
                    tuning_config.output_model_types, base_model.MODEL_TYPE
                ),
            )

```